File: KnapsackProblem.py
def KnapsackProblem(capacity, profit, weight, num):
    if num == 0 or capacity == 0:
        return 0 #base case
    if weight[num - 1] > capacity:
        return KnapsackProblem(capacity, profit, weight, num-1)
    else:
        return max(profit[num-1] + KnapsackProblem(capacity-weight[num-1], profit, weight, num-1), KnapsackProblem(capacity, profit, weight, num-1))

profit = [60, 100, 120]
weight = [10, 20, 30]
capacity = 50
num = len(profit)
print(KnapsackProblem(capacity, profit, weight, num))

File: KnapsackProblem.cpp
#include<iostream>
using namespace std;

int KnapsackProblem(int profit[], int weight[], int capacity, int num)
{
    if (num == 0 || capacity == 0)
    {
        return 0;
    }
    if (weight[num - 1] > capacity)
    {
        return KnapsackProblem(profit, weight, capacity, num - 1);
    }
    else
    {
        return max(KnapsackProblem(profit, weight, capacity - weight[num-1], num-1)+profit[num-1], KnapsackProblem(profit, weight, capacity, num-1));
    }
}

int main(){
    int profit[] = {60, 100, 120};
    int weight[] = {10, 20, 30};
    int capacity = 50;
    int num = 3;
    
    int result = KnapsackProblem(profit, weight, capacity, num);
    cout << "Result: " << result << endl;

    return 0;
}